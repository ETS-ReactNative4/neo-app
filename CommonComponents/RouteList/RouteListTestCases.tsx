import React from "react";

import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import RouteList from "./RouteList";
import { StyleSheet } from "react-native";
import {
  CONSTANT_fontCustom,
  CONSTANT_primaryRegular
} from "../../constants/fonts";

const styles = StyleSheet.create({
  containerStyle: {
    margin: 24
  },
  textStyle: {
    ...CONSTANT_fontCustom(CONSTANT_primaryRegular, 12, 15),
    marginHorizontal: 6
  }
});

const RouteListTestCases: ITestCase[] = [
  {
    title: "Route List",
    Component: (
      <RouteList
        cities={[
          { cityName: "Interlaken" },
          { cityName: "Zerma" },
          { cityName: "Lucerne" }
        ]}
      />
    )
  },
  {
    title: "Route List with custom font styles",
    Component: (
      <RouteList
        cities={[
          { cityName: "Interlaken" },
          { cityName: "Zerma" },
          { cityName: "Lucerne" }
        ]}
        containerStyle={styles.containerStyle}
        routeListTextStyle={styles.textStyle}
      />
    )
  }
];

export default RouteListTestCases;
