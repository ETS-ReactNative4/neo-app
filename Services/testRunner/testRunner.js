import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
