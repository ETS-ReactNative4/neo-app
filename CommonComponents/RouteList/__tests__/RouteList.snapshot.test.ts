import RouteListTestCases from "../RouteListTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

RouteListTestCases.map(testRunner);
