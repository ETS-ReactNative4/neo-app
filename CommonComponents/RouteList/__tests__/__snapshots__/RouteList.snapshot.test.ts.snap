// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route List 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(170, 170, 170, 1)",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "voyager",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 15,
            "marginHorizontal": 8,
          },
          undefined,
        ]
      }
    >
      Interlaken
    </Text>
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(119, 119, 119, 1)",
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 15,
            "marginHorizontal": 8,
          },
          undefined,
        ]
      }
    >
      Zerma
    </Text>
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(119, 119, 119, 1)",
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 15,
            "marginHorizontal": 8,
          },
          undefined,
        ]
      }
    >
      Lucerne
    </Text>
  </View>
</View>
`;

exports[`Route List with custom font styles 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 16,
      },
      Object {
        "margin": 24,
      },
    ]
  }
>
  <Unknown
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(170, 170, 170, 1)",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "voyager",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Unknown>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 15,
            "marginHorizontal": 8,
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "lineHeight": 15,
            "marginHorizontal": 6,
          },
        ]
      }
    >
      Interlaken
    </Unknown>
    <View>
      <Unknown
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(119, 119, 119, 1)",
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Unknown>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 15,
            "marginHorizontal": 8,
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "lineHeight": 15,
            "marginHorizontal": 6,
          },
        ]
      }
    >
      Zerma
    </Unknown>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 15,
            "marginHorizontal": 8,
          },
          Object {
            "color": "rgba(170, 170, 170, 1)",
            "marginHorizontal": 0,
          },
          Object {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "lineHeight": 15,
            "marginHorizontal": 6,
          },
        ]
      }
    >
      +
      1
       
      city
    </Unknown>
  </View>
</View>
`;
